closed/make/Docs.gmk

```diff
@@ -20,4 +20,115 @@
+  CMD_OUTPUT = $(shell if [ -f $(OPENJDK_SUN_MGMT_SRCDIR)/package-info.java ]; then $(MV) $(OPENJDK_SUN_MGMT_SRCDIR)/package-info.java $(OPENJDK_SUN_MGMT_SRCDIR)/package-info.javaz; fi)
+  # Then rename any remaining files we don't want to generate javadoc for
+  # This finds all files ending in .java and then renames each one with an 'x' added add its last character.
+  CMD_OUTPUT = $(shell find $(OPENJDK_SUN_MGMT_SRCDIR) -maxdepth 1 -type 'f' -name "*.java" | xargs -I {} sh -c 'set -x; $(MV) $$1 $${1}x; set +x' -- {})
+  # HotSpotDiagnosticMXBean.java is just a stub class in OpenJ9, so don't generate javadoc for it.
+  CMD_OUTPUT = $(shell if [ -f $(OPENJ9_SUN_MGMT_SRCDIR)/HotSpotDiagnosticMXBean.java ]; then $(MV) $(OPENJ9_SUN_MGMT_SRCDIR)/HotSpotDiagnosticMXBean.java $(OPENJ9_SUN_MGMT_SRCDIR)/HotSpotDiagnosticMXBean.javax; fi)
+  # Finally restore the .java extension of the files we do want to generate javadoc for and the package-info.java file if present
+  CMD_OUTPUT = $(shell find $(OPENJDK_SUN_MGMT_SRCDIR) -maxdepth 1 -type 'f' -name "*.javaz" | xargs -I {} sh -c 'set -x; $(MV) $$1 $${1%?}; set +x' -- {})
+endif
+
+# Target to change the web page footers to the required values.
+# This ensures that openj9 and openjdk classes contain the correct copyrights etc.
+openj9-docs-set-javadoc-bottom:
+	# Set footer for openj9 modules, packages and classes
+	cd $(DOCS_OUTPUTDIR)/api; find . -type 'f' | grep "/openj9" | grep ".html" | xargs -I {} $(OPENJ9_SUBST_COMMAND) {}
+	# Set footer for com.ibm packages and classes
+	cd $(DOCS_OUTPUTDIR)/api; find . -type 'f' | grep "/com/ibm" | grep ".html" | xargs -I {} $(OPENJ9_SUBST_COMMAND) {}
+	# Deal with the com.sun.management classes reimplemented in OpenJ9
+	cd $(DOCS_OUTPUTDIR)/api; find . -type 'f' | grep "GarbageCollectionNotificationInfo.html" | xargs -I {} $(OPENJ9_SUBST_COMMAND) {}
+	cd $(DOCS_OUTPUTDIR)/api; find . -type 'f' | grep "GcInfo.html" | xargs -I {} $(OPENJ9_SUBST_COMMAND) {}
+	# Set footer for all other files to the openjdk footer
+	cd $(DOCS_OUTPUTDIR)/api; find . -type 'f' | xargs -I {} $(OPENJDK_SUBST_COMMAND) {}
+
+openj9-docs-restore-files:
+	# This finds all files ending in .javax and then renames each one with its last character removed.
+	find $(OPENJDK_SUN_MGMT_SRCDIR) -maxdepth 1 -type 'f' -name "*.javax" | xargs -I {} sh -c 'set -x; $(MV) $$1 $${1%?}; set +x' -- {}
+	find $(OPENJ9_SUN_MGMT_SRCDIR) -maxdepth 1 -type 'f' -name "*.javax" | xargs -I {} sh -c 'set -x; $(MV) $$1 $${1%?}; set +x' -- {}
+	# Also restore any files ending in .javaz which would exist if a previous run was interrupted in the small window when those files were renamed
+	find $(OPENJDK_SUN_MGMT_SRCDIR) -maxdepth 1 -type 'f' -name "*.javaz" | xargs -I {} sh -c 'set -x; $(MV) $$1 $${1%?}; set +x' -- {}
+
+PHONY: openj9-docs-set-javadoc-bottom openj9-docs-restore-files
```

Comment:
This needs to be `.PHONY` (the `.` is missing).

Reply:
Well spotted.   Thanks.